monai/networks/utils.py

```diff
@@ -93,7 +93,13 @@ def one_hot(labels: torch.Tensor, num_classes: int, dtype: torch.dtype = torch.f
     return labels
 
 
+@deprecated(since="0.8.0", msg_suffix="use `monai.utils.misc.sample_slices` instead.")
 def slice_channels(tensor: torch.Tensor, *slicevals: Optional[int]) -> torch.Tensor:
+    """
+    .. deprecated:: 0.8.0
```

Comment:
Is this existing utility good enough for @holgerroth's request?

Reply:
Hi @wyli ,

I think it's not good enough for @holgerroth 's request.
The `slicevals` arg of this existing function is slightly misleading, it actually defines the `start` and `end`, etc. for `slice` function of python. For example:
```py
from monai.networks import slice_channels


data = torch.tensor([[[0, 1], [2, 3], [4, 5]]])
ret = slice_channels(data, 0, 2)
print(ret)
```
Will output:
```
tensor([[[0, 1], [2, 3]]])
```
@holgerroth asked to select specified classes like [1, 4, 6, 8] instead of continuous slices, the new utility in this PR accepts index list of classes and supports to specify the `dim`.

Thanks.

Reply:
ok, that's good, would also be great to have a flag to keep the `slice(*slicevals)`as the syntax would be simpler for 'all but the first channel' `[1:]`.

Reply:
Hi @wyli ,

Actually, the previous utility doesn't support slice like `[1:]`, it must specify the `start` and `end` indices, for example:
```py
from monai.networks import slice_channels


data = torch.tensor([[[0, 1], [2, 3], [4, 5]]])
ret = slice_channels(data, 1)  # use 1 to test [1: ]
print(ret)
```
The output is: `tensor([[[0, 1]]])` instead of expected `torch.tensor([[[2, 3], [4, 5]]])`.
So could you please help share more details of what you expect the flag to do?

Thanks in advance.

Reply:
yes, it also supports the slicing method such as
```py
data = torch.zeros((2, 3, 4, 5, 6))
slice_channels(data, 1, None).shape
slice_channels(data, 1, -1).shape
```
maybe it's useful in some cases for the configuration based workflows...

Reply:
OK, sure, added the flag in the latest commit.
Could you please help review it again?

Thanks.